parquet-column/src/main/java/org/apache/parquet/io/MessageColumnIO.java

```diff
@@ -399,6 +444,12 @@ public void addDouble(double value) {
       if (DEBUG) printState();
     }
 
+
+    //should flush null for all groups
+    @Override
+    public void flush() {
+      flushCachedNulls(MessageColumnIO.this);
```

Comment:
is this called at the end of the row group (or page?) to make sure any dangling nulls are properly flushed? 
Do we have a unit test to cover the scenario where the values are:
```
(some non-nulls)....(some nulls)...(some non-nulls)...(some nulls)(end of stream)
----------------------------------------------------------------------^ verify that these trailing nulls are written
```